packages/core/src/controllers/verify.ts

```diff
@@ -161,4 +167,46 @@ export class Verify extends IVerify {
     }
     return url;
   };
+
+  private fetchPublicKey = async () => {
+    this.logger.info(`fetching public key from: ${this.verifyUrlV2}`);
+    const timeout = this.startAbortTimer(FIVE_SECONDS);
+    const result = await fetch(`${this.verifyUrlV2}/public-key`, {
+      signal: this.abortController.signal,
+    });
+    clearTimeout(timeout);
+    return (await result.json()) as jwk;
+  };
+
+  private persistPublicKey = async (publicKey: jwk) => {
+    console.log(`persisting public key to local storage`, publicKey);
+    await this.store.setItem(this.storeKey, publicKey);
+    this.publicKey = publicKey;
+  };
+
+  private removePublicKey = async () => {
+    console.log(`removing public key from local storage`);
+    await this.store.removeItem(this.storeKey);
+    this.publicKey = undefined;
+  };
+
+  private isValidJwtAttestation = async (attestation: string) => {
+    if (!this.publicKey) {
+      console.log("public key not found");
+      return;
+    }
+    const cryptoKey = await getCryptoKeyFromKeyData(this.publicKey.publicKey);
```

Comment:
If this fails verification it should request the latest key: https://www.notion.so/walletconnect/Verify-Server-rewrite-tech-doc-Verify-API-V2-30e207c77f8447d8af37a186799b7c9e?pvs=4#b681b45f31384899b46985bff30266b4